Tx list tx details

DiogoSoaress opened the pull request:
## What it solves
Resolves #2989

## How this PR fixes it
Now it should be possible to verify advanced Tx data in all types of transactions:

- safeTxHash
- nonce
- to
- value
- raw data - no need to display in full, being able to copy is sufficient
- operation
- safeTxGas
- baseGas
- gasPrice
- gasToken
- refundReceiver
- signatures

## How to test it
Go to the Transactions list.
Click on one of the entries.
The transactions details (queued/history) shall display the Tx advanced info

## Screenshots
![Screen Shot 2021-12-16 at 11 09 28](https://user-images.githubusercontent.com/32431609/146352276-f68ba392-8f96-43ad-927c-afa497708d4d.png)
![Screen Shot 2021-12-16 at 11 10 06](https://user-images.githubusercontent.com/32431609/146352279-0c99b5ca-e4ba-4bba-a1c1-68bc95de7431.png)
![Screen Shot 2021-12-16 at 11 10 20](https://user-images.githubusercontent.com/32431609/146352281-fe8a7279-f2d9-4bc8-8188-9c07c9fa86d0.png)
![Screen Shot 2021-12-16 at 11 11 10](https://user-images.githubusercontent.com/32431609/146352282-1f5d110f-bed4-4352-a049-9834ceb0065d.png)

DiogoSoaress commented:
The PR addresses some major visual changes.
The ticket https://github.com/gnosis/safe-react/issues/3067 still stands however for some of the UI enhancements.

iamacook commented on src/logic/safe/store/actions/createTransaction.ts:
```diff
@@ -126,7 +126,7 @@ export const createTransaction =
 
     let txHash
-    const txArgs: TxArgs & { sender: string } = {
+    const txArgs: TxArgs = {
```
Why have you remove `sender` from the arguments?

DiogoSoaress replied:
I made it mandatory in `TxArgs` instead.

iamacook replied:
Bizarre. I guess a transaction **always** has a sender!

iamacook commented on src/logic/safe/store/models/types/transaction.ts:
```diff
@@ -17,7 +17,7 @@ export type TxArgs = {
   safeInstance: GnosisSafe
   safeTxGas: string
-  sender?: string
+  sender: string
```
Why is this now required?

DiogoSoaress replied:
Because we were setting it to required by using `intersection types` to make it required

iamacook commented on src/logic/safe/transactions/txHistory.ts:
```diff
@@ -2,13 +2,19 @@ import { GnosisSafe } from 'src/types/contracts/gnosis_safe.d'
 
 import { checksumAddress } from 'src/utils/checksumAddress'
-import { proposeTransaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { MultisigTransactionRequest, proposeTransaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
```
You should move this import to the top with a space below it.

iamacook commented on src/logic/safe/transactions/txHistory.ts:
```diff
@@ -2,13 +2,19 @@ import { GnosisSafe } from 'src/types/contracts/gnosis_safe.d'
 
-const calculateBodyFrom = async (
-  safeInstance: GnosisSafe,
+type ProposeTxBodyProps = Omit<MultisigTransactionRequest, 'safeTxHash'> & {
```
This is not a component. They are not 'props'.

iamacook commented on src/logic/safe/transactions/txHistory.ts:
```diff
@@ -2,13 +2,19 @@ import { GnosisSafe } from 'src/types/contracts/gnosis_safe.d'
+  safeInstance,
   to,
-  valueInWei,
+  value,
```
I wouldn't change the argument name. It is clearer as `valueInWei`.

DiogoSoaress replied:
In this case I tried to be coherent with the types that come from `type MultisigTransactionRequest` . Plus, it is not an exported function so it doesn't change much.

iamacook replied:
This also affects the `getTransactionHash()` method though.

DiogoSoaress replied:
But here we are calling the contract method directly, not the `getTransactionHash()`

DiogoSoaress replied:
I  just wanted the methods to be typed . Can leave the names as before it it causes confusion

iamacook replied:
On line 31, you pass the `value` to the Safe instance.

iamacook replied:
You can leave it as is, but I would personally refrain from changes like this in the future. I just feel like it removes semantic clarity.

iamacook commented on src/logic/safe/transactions/txHistory.ts:
```diff
@@ -43,10 +49,7 @@ const calculateBodyFrom = async (
-  safeInstance: GnosisSafe
-  [key: string]: any
-}
+type SaveTxToHistoryProps = TxArgs & { origin?: string | null; signature?: string }
```
This is not a component. They are not 'props'.

iamacook commented on src/logic/safe/transactions/txHistory.ts:
```diff
@@ -63,24 +66,24 @@ export const saveTxToHistory = async ({
     data,
     operation,
-    nonce,
+    nonce: nonce.toString(),
```
Why have you changed this to a string?

DiogoSoaress replied:
Because of the `MultisigTransactionRequest` types. It expects `nonce` as a string.

if I leave it as `number`...
> The expected type comes from property 'nonce' which is declared here on type 'ProposeTxBodyTypes'

iamacook replied:
I get that you're adhering to the types but you added the types yourself (`ProposeTxBodyProps`) in this PR. What's to say it was not expecting a number previously? Have you tested this?

DiogoSoaress replied:
I've tested it.
I made the return type in `calculateBodyFrom` to be `MultisigTransactionRequest` because it is the type used when you pass `body` to
<img width="842" alt="Screen Shot 2021-12-17 at 14 33 40" src="https://user-images.githubusercontent.com/32431609/146552481-6c247c73-cce2-4979-aa35-27ebeb22a91b.png">
 
 so I believe it is the correct type
 I believe

iamacook commented on src/routes/safe/components/Transactions/TxList/TxDetails.tsx:
```diff
@@ -106,31 +106,37 @@ export const TxDetails = ({ transaction, actions }: TxDetailsProps): ReactElemen
+          className={cn('tx-details', {
+            'no-padding': isMultiSendTxInfo(data.txInfo),
+            'not-executed': !data.executedAt,
+            'will-be-replaced': transaction.txStatus === 'WILL_BE_REPLACED',
```
A nice way to clean this us a bit is to move some of these conditions into variables as they are repeated, e.g.

```ts
const willBeReplaced = transaction.txStatus === 'WILL_BE_REPLACED'
const isHistorical = txLocation === 'history'
```

iamacook commented on src/routes/safe/components/Transactions/TxList/TxDetails.tsx:
```diff
@@ -106,31 +106,37 @@ export const TxDetails = ({ transaction, actions }: TxDetailsProps): ReactElemen
-        })}
-      >
-        <TxOwners txDetails={data} />
+      <div>
```
Have you checked all forms of transactions after moving the order of these?

DiogoSoaress replied:
I checked for all the transactions that we have in the task description. I think I've got them all but the ticket will be scrutinized by QA anyway. So far seems to work everywhere.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxExpandedActions.tsx:
```diff
@@ -41,6 +41,11 @@ export const TxExpandedActions = ({ transaction }: TxExpandedActionsProps): Reac
   // https://github.com/facebook/react/issues/4492
   return (
     <>
+      {canCancel && (
```
Did you also change the collapsed actions to reflect this reordering?

DiogoSoaress replied:
I will. Excellent point!

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx:
```diff
@@ -21,7 +26,7 @@ export const TxInfoCreation = ({ transaction }: { transaction: Transaction }): R
     <TxDetailsContainer>
       <div className="tx-summary">
-        <div className="tx-hash">
+        <StyledGridRow>
```
Why remove the class names?

DiogoSoaress replied:
In this file (which is only for the Creation Tx) I only removed the classes that were not being targeted or that were not relevant for the new design.

iamacook replied:
Don't forget E2E tests...

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx:
```diff
@@ -21,7 +26,7 @@ export const TxInfoCreation = ({ transaction }: { transaction: Transaction }): R
     <TxDetailsContainer>
       <div className="tx-summary">
-        <div className="tx-hash">
+        <StyledGridRow>
```
It seems like every `<StyledGridRow>` contains a `<Text>` 'title'. You could move that into the component as pass it as a prop. The `children` could then be the 'value'.

DiogoSoaress replied:
The `StyledGridRow` is just a styled div. Is it worth making a component out of it?

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx:
```diff
@@ -96,9 +99,10 @@ export const TxInfoCreation = ({ transaction }: { transaction: Transaction }): R
+        </StyledGridRow>
       </div>
-      <div className="tx-owners" />
+      <StyledPadding></StyledPadding>
```
Dead code.

DiogoSoaress replied:
It is a "filler" as the parent is a grid with 2 columns. Let me know if there's a better way do fill it.

iamacook replied:
I'd add a brief comment. You need not include an opening and closing tag as well, e.g. `<StyledPadding />`

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx:
```diff
@@ -96,9 +99,10 @@ export const TxInfoCreation = ({ transaction }: { transaction: Transaction }): R
       </div>
-      <div className="tx-owners" />
+      <StyledPadding></StyledPadding>
+      {/* <div className="tx-owners" /> */}
```
Unnecessary comment.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoMultiSend.tsx:
```diff
@@ -20,6 +22,12 @@ const TxInfoMultiSend = ({ txInfo }: { txInfo: MultiSend }): ReactElement => {
         showCopyBtn
         explorerUrl={getExplorerInfo(hash)}
       />
+      <Text size="xl" strong as="span">
```
You could use your `<StyledGridRow>` here as well, no?

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoMultiSend.tsx:
```diff
@@ -1,5 +1,6 @@
 import { ReactElement } from 'react'
 import { MultiSend } from '@gnosis.pm/safe-react-gateway-sdk'
```
External imports to the top.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx:
```diff
@@ -5,10 +5,15 @@ import { getExplorerInfo } from 'src/config'
 import { Creation } from '@gnosis.pm/safe-react-gateway-sdk'
 import { useKnownAddress } from './hooks/useKnownAddress'
 import PrefixedEthHashInfo from 'src/components/PrefixedEthHashInfo'
+import styled from 'styled-components'
```
External imports to the top.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
 import { IS_PRODUCTION } from 'src/utils/constants'
 import TxInfoMultiSend from './TxInfoMultiSend'
 import DelegateCallWarning from './DelegateCallWarning'
+import styled from 'styled-components'
```
External imports to the top.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
 import TxInfoMultiSend from './TxInfoMultiSend'
 import DelegateCallWarning from './DelegateCallWarning'
+import styled from 'styled-components'
+import { getByteLength } from 'src/components/DecodeTxs' // must pass to utils
```
Do you intend to move this function to `/utils`?

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
+    {inlineType === 'rawData' && (
+      <FlexWrapper margin={5}>
+        <Text size="xl">{value ? getByteLength(value) : 0} bytes</Text>
+        <CopyToClipboardBtn textToCopy={value || ''} />
```
I'd suggest conditionally adding this instead of allowing an empty string to be copied, e.g.

```ts
{value && <CopyToClipboardBtn textToCopy={value} />}
```

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
 
 type Props = { txDetails: ExpandedTxDetails }
 
+const FlexWrapper = styled.div<{ margin: number }>`
```
Can you add a comment if this reacts to the class name-based styles within the transaction list? I think we should definitely rely purely on `styled-components` so it's nice to see this.

DiogoSoaress replied:
Can you explain in more detail the first part of the comment?

iamacook replied:
This was just a comment. I like that you used `styled-components` here 😃

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
+    <Text size="xl" as="span">
+      {title}
+    </Text>
+    {inlineType === 'hash' && (
```
It's definitely personal preference, but it may be a bit more readable if you moved this into a getter function outside of the render. That way you can use a `switch`, e.g.

```ts
const TxDataRow = (props) => {
  const getInlineType = () => {
    switch (props.inlineType) {
      case 'hash': {
        return ...
      }
      ...
    }
  }

  return (
    ...
    getInlineType()
  )
}
```

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
+
+  let created, confirmations, safeTxHash, baseGas, gasPrice, gasToken, refundReceiver, safeTxGas
+  if (isMultiSigExecutionDetails(detailedExecutionInfo)) {
+    // prettier-ignore
```
Don't ignore, be it with Prettier or TypeScript.

DiogoSoaress replied:
It kept adding a `;` before the parenthesis, breaking the destructuring... Can we leave this one?

iamacook replied:
It's defining a new line with the `;`. If it's that much of a pain, leave it.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
+
+  let signaturesFromConfirmations
+  if (confirmations && confirmations?.length > 0) {
+    const signatures = confirmations?.map(({ signer, signature }) =>
```
You need not optionally chain here.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
-            {nonce}
-          </Text>
-        </div>
+      {txHash && <TxDataRow title="Transaction hash:" value={txHash} inlineType="hash" />}
```
Noice refactoring!

iamacook replied:
Why remove the class names though?

iamacook replied:
It might be used in E2E tests?

DiogoSoaress replied:
I've just checked. More, it would have broken the E2E in the PR.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
+      {safeTxHash && <TxDataRow title="SafeTxHash:" value={safeTxHash} inlineType="hash" />}
+      {created && <TxDataRow title="Created:" value={formatDateTime(created)} />}
+      <TxDataRow title="Executed:" value={executedAt ? formatDateTime(executedAt) : NOT_AVAILABLE} />
+      <br></br>
```
I would only use `<br>` for breaking a line of text. This can be done with CSS.

DiogoSoaress replied:
Can you suggest a snippet here? I see we are using `<br />` in more places that's why I didn't hesitate in also using it here

iamacook replied:
Add a wrapper with a margin-bottom?

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
+
+      {/* Advanced TxData */}
+      {txData?.operation !== undefined && (
+        <TxDataRow title="Operation:" value={`${txData?.operation} (${Operation[txData?.operation].toLowerCase()})`} />
```
You need not optional chain here. It will prevent later type warnings if the condition was removed.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
-            {formatDateTime(created)}
-          </Text>
-        </div>
+      {safeTxGas && <TxDataRow title="safeTxGas:" value={safeTxGas} />}
```
Maybe you should use proper terms instead of keys here, e.g. 'Gas price:'

DiogoSoaress replied:
Let's comment it on the ticket so Designs can reflect it.

iamacook replied:
Why can't you just adjust the text here? It's a label.

DiogoSoaress replied:
I understand it is an easy tweak. My point is there are more fields where we are using the keys. For now it is coherent with the Mobile implementation reason why I think should have Johannes helping with the renaming and addressing it in both applications (if applicable).

iamacook replied:
I follow you, but some of the keys are just camel cased labels. If you'd sooner shift it to the design ticket, fair enough.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
+      {gasToken && <TxDataRow title="gasToken:" value={gasToken} inlineType="hash" />}
+      {refundReceiver && <TxDataRow title="refundReceiver:" value={refundReceiver} inlineType="hash" />}
+      {confirmations?.map(({ signature }, index) => (
+        <TxDataRow title={`Signature ${index + 1} :`} value={signature} inlineType="rawData" key={index} />
```
Don't use indexes as keys.

iamacook replied:
There shouldn't be a space before a colon. (The same goes for the following line.)

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -10,21 +10,77 @@ import {
+  }
+
+  let signaturesFromConfirmations
+  if (confirmations && confirmations?.length > 0) {
```
You need not check both, `confirmations?.length > 0` should be sufficient.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
-          {executedAt ? formatDateTime(executedAt) : NOT_AVAILABLE}
-        </Text>
-      </div>
+      {txData?.hexData && <TxDataRow title="Raw data:" value={txData?.hexData} inlineType="rawData" />}
```
No need for optional chaining. It would break later type checking.

DiogoSoaress replied:
Oh I understand what you mean!

iamacook commented on src/routes/safe/components/Transactions/TxList/TxSummary.tsx:
```diff
@@ -33,54 +89,28 @@ export const TxSummary = ({ txDetails }: Props): ReactElement => {
+      {txData?.hexData && <TxDataRow title="Raw data:" value={txData?.hexData} inlineType="rawData" />}
       {txData?.operation === Operation.DELEGATE && (
         <div className="tx-operation">
           <DelegateCallWarning isKnown={isCustomTxInfo(txInfo) && !!txInfo?.to?.name} />
```
Does the delegate warning fit with the new style adjustments? It is also present on nested delegate calls.

DiogoSoaress replied:
It seems to be right. Might be edges cases that I didn't test
<img width="775" alt="Screen Shot 2021-12-17 at 14 53 57" src="https://user-images.githubusercontent.com/32431609/146554782-5a68639b-8cdf-4a40-af02-01932dff1375.png">

iamacook commented on src/routes/safe/components/Transactions/TxList/styled.tsx:
```diff
@@ -378,24 +384,23 @@ export const TxDetailsContainer = styled.div`
   .tx-owners {
     padding: 24px;
-    grid-column-start: 2;
+    /* grid-column-start: 2;
```
Remove comments.

iamacook commented on src/routes/safe/components/Transactions/TxList/styled.tsx:
```diff
@@ -378,24 +384,23 @@ export const TxDetailsContainer = styled.div`
 
     &.no-owner {
-      grid-row-end: span 3;
+      /* grid-row-end: span 3; */
```
Have you tested this? This would be unstyled.

DiogoSoaress replied:
I've tested and removed the class as it became obsolete.

iamacook reviewed (commented):
Looking good so far!

iamacook commented:
@DiogoSoaress, further to my review:

> ![Screen Shot 2021-12-16 at 11 09 28](https://user-images.githubusercontent.com/32431609/146352276-f68ba392-8f96-43ad-927c-afa497708d4d.png)

The alignment within the collapse doesn't seem quite right to me with the shift from middle to right, especially in the screenshot below. The number of confirmations should be above the owners imo. @liliiaorlenko

> ![Screen Shot 2021-12-16 at 11 11 10](https://user-images.githubusercontent.com/32431609/146352282-1f5d110f-bed4-4352-a049-9834ceb0065d.png)

The owner panel should be made a bit wider or add a line break after 'Execute'. Otherwise it will always split 'needed'.

DiogoSoaress commented:
> @DiogoSoaress, further to my review:
> 
> > ![Screen Shot 2021-12-16 at 11 09 28](https://user-images.githubusercontent.com/32431609/146352276-f68ba392-8f96-43ad-927c-afa497708d4d.png)
> 
> The alignment within the collapse doesn't seem quite right to me with the shift from middle to right, especially in the screenshot below. The number of confirmations should be above the owners imo. @liliiaorlenko
> 
> > ![Screen Shot 2021-12-16 at 11 11 10](https://user-images.githubusercontent.com/32431609/146352282-1f5d110f-bed4-4352-a049-9834ceb0065d.png)
> 
> The owner panel should be made a bit wider or add a line break after 'Execute'. Otherwise it will always split 'needed'.

Aaron I opted to not address all the design rework to the detail as there is a dedicated task for that https://github.com/gnosis/safe-react/issues/3067 . When that task is finished, our UI should resemble the one from the mockups.

iamacook commented:
> Aaron I opted to not address all the design rework to the detail as there is a dedicated task for that #3067 . When that task is finished, our UI should resemble the one from the mockups.

That's completely understandable. However, seeing as it was adjusted in this PR, can you move the vertical divider a bit more left to avoid the execute line break?

iamacook commented on src/routes/safe/components/Transactions/TxList/TxDataRow.tsx:
```diff
@@ -0,0 +1,41 @@
+  }
+`
+
+type TxDataRowType = { children?: ReactNode; inlineType?: 'hash' | 'rawData'; title: string; value: string | null }
```
Why don't you just make `value` optional? You won't need to check the type each time.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxDataRow.tsx:
```diff
@@ -0,0 +1,41 @@
+    {inlineType === 'rawData' && typeof value === 'string' && (
+      <FlexWrapper margin={5}>
+        <Text size="xl">{value ? getByteLength(value) : 0} bytes</Text>
+        {value && <CopyToClipboardBtn textToCopy={value || ''} />}
```
You don't need the empty string fallback.

iamacook commented on src/routes/safe/components/Transactions/TxList/TxInfoCreation.tsx:
```diff
@@ -21,32 +27,9 @@ export const TxInfoCreation = ({ transaction }: { transaction: Transaction }): R
-          </Text>
+        <TxDataRow title="Transaction hash:" value={txInfo.transactionHash} inlineType="hash" />
+        <TxDataRow title="Created:" value={formatDateTime(timestamp)} />
+        <TxDataRow title="Creator:" value={null}>
```
What's with the `null` value? I think this shows why it being optional makes sense.

iamacook reviewed (commented):
I think we need to rethink the `<TxDataRow>` `value` prop.

DiogoSoaress commented:
> I think we need to rethink the `<TxDataRow>` `value` prop.

I removed the `null` type and made the `value` prop optional 👍  Also there were some unnecessary typechecks in the if conditions so I also cleaned them up.

francovenica commented:
SafeTxHash should not have to show the "View in blockExplorer" button
![image](https://user-images.githubusercontent.com/4503659/146762605-0a5e2028-5a5a-4bab-9c35-f291516fe6f7.png)

DiogoSoaress commented:
@francovenica I've removed the "explorer" for the `safeTxHash` so please give it another round.

francovenica commented:
I'm checking tx in xDai. The ones that send xdai tokens are not showing "Raw data", even tho in the block explorer there is "Raw data"
Safe: https://pr3184--safereact.review-safe.gnosisdev.com/app/xdai:0xB8d760a90a5ed54D3c2b3EFC231277e99188642A/transactions/history#prod

![image](https://user-images.githubusercontent.com/4503659/146778888-3c1efd30-8b62-43fc-918c-b3279f583fa2.png)

https://blockscout.com/xdai/mainnet/tx/0x0bb58626d398f32e36683144e4d9f210a697469adae8eefcbda8581659ad009d/
![image](https://user-images.githubusercontent.com/4503659/146778897-359b0e03-4c83-400b-a2ed-5fde79a336d2.png)

Other tx they do show Raw data just fine
![image](https://user-images.githubusercontent.com/4503659/146779028-91d3c17f-67db-45d9-b0f4-ec05e1f35ec3.png)

francovenica commented:
Questions: (not issues, just looking for clarification)
1 - The Receive type of tx show little to no info. I assume is this way because is more important tx that go out of the safe. Still I want to be sure this is the case

2 - In every tx I've seen so far, in different networks even there are values that are always the same. so, are these even useful? Is not they are wrong, you can see the same values in the block explorer in the tx, but since those are always 0 idk how much use they can give.
![image](https://user-images.githubusercontent.com/4503659/146799042-609c092a-c680-4f05-97e1-a54ed9a45278.png)

francovenica commented:
Issue:
The Safe creation tx looks off:
How it looks now:
![image](https://user-images.githubusercontent.com/4503659/146799356-f40577d5-1167-482e-8195-13d530633d26.png)

How it used to look:
![image](https://user-images.githubusercontent.com/4503659/146799377-e9c2632e-ba0d-47c0-8fa8-c7efd7cc6fb8.png)

francovenica commented:
In general it looks good. 

The only issues are the lack of raw data in tx of other networks that are not ethereum and the safe create tx that looks kinda off
Then there are the questions in this comment https://github.com/gnosis/safe-react/pull/3184#issuecomment-998076307

cc @DiogoSoaress @johannesmoormann

DiogoSoaress commented:
@francovenica I will check why some fields are missing in some networks (xdai & ewc). This ticket aims to display more advanced tx data and not to have the final designs. We will merge this PR it to the epic branch once QAed and not to `dev`.

Regarding the layouts, they will be improved in https://github.com/gnosis/safe-react/issues/3067 to be picked next to this ticket. Can you provide a screen for  "Safe Creation" transaction @liliiaorlenko ?

DiogoSoaress commented:
> I'm checking tx in xDai. The ones that send xdai tokens are not showing "Raw data", even tho in the block explorer there is "Raw data". Maybe because is called "Raw input" in the block explorer? are raw data and raw input maybe different things?
> 
> EDIT: same for EWC here.
> 
> xdai Safe: https://pr3184--safereact.review-safe.gnosisdev.com/app/xdai:0xB8d760a90a5ed54D3c2b3EFC231277e99188642A/transactions/history#prod
> 
> EWC safe: https://pr3184--safereact.review-safe.gnosisdev.com/app/ewt:0x082fcc9f1A997b45395aBf7FD7DEeA2D0Ce68270/transactions/history (check latest tx in the history tab)
> 
> ![image](https://user-images.githubusercontent.com/4503659/146778888-3c1efd30-8b62-43fc-918c-b3279f583fa2.png)
> 
> https://blockscout.com/xdai/mainnet/tx/0x0bb58626d398f32e36683144e4d9f210a697469adae8eefcbda8581659ad009d/ ![image](https://user-images.githubusercontent.com/4503659/146778897-359b0e03-4c83-400b-a2ed-5fde79a336d2.png)
> 
> Other type of tx they do show Raw data just fine ![image](https://user-images.githubusercontent.com/4503659/146779028-91d3c17f-67db-45d9-b0f4-ec05e1f35ec3.png)

I was checking the behaviour in these 2 cases and the backend is answering with `txData.hexData = null` for xDai and EWC so I think we should ask the BE team in this regard
_for example in:_
https://safe-client.gnosis.io/v1/chains/246/transactions/multisig_0x082fcc9f1A997b45395aBf7FD7DEeA2D0Ce68270_0x4198f0a11ecafbd1871abae2670edcabe2f0911cf2ced1064eadfa5d8a63623a
AND
https://safe-client.gnosis.io/v1/chains/100/transactions/multisig_0xB8d760a90a5ed54D3c2b3EFC231277e99188642A_0x7a17f0e727470a01c198d1c78952229d24801003c56bd0bf19feedf631231c34

DiogoSoaress commented:
@francovenica so the clarification I got is that most of native coin transfers don't have `data` .  Other transactions that not Send Funds should contain the Raw Data. However, this field is nullable and might not appear in simple send funds.

francovenica commented:
Ok, so the raw data not showing in other networks is normal behavior.

francovenica commented:
Ok, I've given the responses from the question
The values that always show 0's and the 0x000....000 values will remain there, since mobile is doing the same, so we keep consistency

Regarding "Receive" tx, the reason that those are not showing more info is because the backend is not providing more info
![image](https://user-images.githubusercontent.com/4503659/146968277-573bf540-c9a9-4f5d-bf19-5e90cf4d0b23.png)

This looks fine, we can merge it into the epic.
